src/annotationLayer.js

```diff
@@ -348,7 +358,7 @@ var annotationLayer = function (args) {
           state: geo_annotation.state.create,
           layer: this
         });
-        this.addAnnotation(m_this.currentAnnotation);
+        this.addAnnotation(m_this.currentAnnotation, null);
```

Comment:
it could be little confusing in the way we are using null or undefined. I would suggest that if it is null or undefined, we fall back to map.gcs(). Should we create keys perhaps? map_gcs, map_ingcs? Thoughts?